### PR TITLE
Add docs to webhook subscriptions

### PR DESCRIPTION
**Pull request by jakubkuc**
This pr adds documentation to webhook subscription payloads

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]
Maybe we could include a link to the presentation from GraphQLWroc prepared by @korycins?

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: Subscription Webhook Payloads
+---
+
+## Custom payloads
+
+The whole idea behind subscription payloads for webhooks is simple. We want to be able to generate custom payloads for all webhooks using our GraphQL API. With subscriptions we are able to do that.
+Our Webhook objects have a `query` field that can hold subscription query converted to string. That query will be used to build a custom webhook payload.
```
Comment (maarcingebala) [inline docs/developer/extending/apps/subscription-webhook-payloads.mdx:8]: We should avoid saying "simple" in the docs ([see this article for instance](https://www.knowledgeowl.com/home/dont-say-simply-jim-fisher)). I would rephrase this part a bit:

Webhook payloads in Saleor can be defined with GraphQL subscriptions. Subscription queries allow you to subscribe to different events, and for each one, define what fields should be returned in the payload.

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+## Building subscription query
+
+To build subscription query you can use any GraphQL interface (like e.g. Saleor playground). Just like with
+building a GraphQL query you will be able to use prompts and build the whole subscription from our API. Payload generation will use resolvers/permissions that are normally used in our API.
+Inside the query you must define event type for which payload will be generated. Event type must be supported by our subscription mechanism (must be subscribable).
```
Comment (maarcingebala) [inline docs/developer/extending/apps/subscription-webhook-payloads.mdx:14]:
```suggestion
Inside the query, you must define the event type for which the payload will be generated. The event type must be supported by our subscription mechanism (see the [list of supported events](developer/api-reference/unions/event.mdx)).
```

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        "id": "<id>",
+        "type": "user"
+      }
+    }
+  }
+```
+
+## Permissions
+
+Since subscriptions uses Saleor API to define webhook payload, the App that has the webhook subscribed needs all the permissions that would normally be required when querying for specified data. Otherwise payload will not be resolved correctly.
```
Comment (maarcingebala) [inline docs/developer/extending/apps/subscription-webhook-payloads.mdx:92]:
```suggestion
Since subscriptions use Saleor API to define webhook payload, the App that has the webhook subscribed needs all the permissions that would normally be required when querying for specified data. Otherwise, payload will not be resolved correctly.
```

**Comment by maarcingebala** [comment]
@patrys Do you think this topic deserves a separate page in the docs? Eventually, this will become the standard behavior for all webhooks.

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 4]
```diff
@@ -0,0 +1,92 @@
+---
+title: Subscription Webhook Payloads
```
Comment (korycins) [inline docs/developer/extending/apps/subscription-webhook-payloads.mdx:2]: We should include info about the version when we added it and also that this is a previe feature

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  }
+}
+```
+
+
+Payload generated from subscription will have all the fields that you asked for, plus some meta fields.
+
+Example payload:
+
+```graphql
```
Comment (korycins) [inline docs/developer/extending/apps/subscription-webhook-payloads.mdx:73]: It is not graphql, it is an json

**Comment by patrys** [comment]
I think it deserves a page as long as it's in preview. I want to make sure that we recommend it over the previous approach. I'm not too fond of it sitting at the section's end.

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx (hunk 3 above)
Comment (jakubkuc) [reply docs/developer/extending/apps/subscription-webhook-payloads.mdx]: fixed

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx (hunk 2 above)
Comment (jakubkuc) [reply docs/developer/extending/apps/subscription-webhook-payloads.mdx]: fixed

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx (hunk 1 above)
Comment (jakubkuc) [reply docs/developer/extending/apps/subscription-webhook-payloads.mdx]: You're right. Changed to proposed.

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+building a GraphQL query you will be able to use prompts and build the whole subscription from our API. Payload generation will use resolvers/permissions that are normally used in our API.
+Inside the query you must define event type for which payload will be generated. Event type must be supported by our subscription mechanism (must be subscribable).
+
+Events that are currently subscribable can be found in the [List of Events](developer/api-reference/unions/event.mdx)
+
+:::warning
+Only asynchronous webhooks are able to use subscriptions at the moment. Synchronous webhooks will be supported in the future.
+:::
+
+Subscription query example:
```
[the remark this answers is not in the document]
Comment (jakubkuc) [reply docs/developer/extending/apps/subscription-webhook-payloads.mdx]: fixed

**Comment by jakubkuc** [comment]
Changed location in the sidebar to right after async and sync webhooks @patrys . It suits better i think.

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+## Custom payloads
+
+Webhook payloads in Saleor can be defined with GraphQL subscriptions. Subscription queries allow you to subscribe to different events, and for each one, define what fields should be returned in the payload.
+
+## Building subscription query
+
+To build subscription query you can use any GraphQL interface (like e.g. Saleor playground). Just like with
+building a GraphQL query you will be able to use prompts and build the whole subscription from our API. Payload generation will use resolvers/permissions that are normally used in our API.
+Inside the query, you must define the event type for which the payload will be generated. The event type must be supported by our subscription mechanism (see the [list of supported events](developer/api-reference/unions/event.mdx)
```
Comment (patrys) [inline docs/developer/extending/apps/subscription-webhook-payloads.mdx:23]:
```suggestion
Inside the query, you must define the event type for which the payload will be generated. The event type must be supported by our subscription mechanism (see the [list of supported events](developer/api-reference/unions/event.mdx)).
```

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+:::
+
+
+## Custom payloads
+
+Webhook payloads in Saleor can be defined with GraphQL subscriptions. Subscription queries allow you to subscribe to different events, and for each one, define what fields should be returned in the payload.
+
+## Building subscription query
+
+To build subscription query you can use any GraphQL interface (like e.g. Saleor playground). Just like with
```
Comment (patrys) [inline docs/developer/extending/apps/subscription-webhook-payloads.mdx:21]:
```suggestion
To build a subscription query you can use any GraphQL interface (like [GraphQL Playground](developer/getting-started/playground.mdx)). Just like with
```

File: docs/developer/extending/apps/subscription-webhook-payloads.mdx (hunk 7 above)
Comment (jakubkuc) [reply docs/developer/extending/apps/subscription-webhook-payloads.mdx]: fixed

**Comment by patrys** [comment]
@jakubkuc Please fix the CLA issue (or update the commit with the correct email address).

**Review by IKarbowiak — APPROVED** [review]

**Comment by patrys** [comment]
I'm merging this, it's good enough, and we can improve it after publishing the initial version.

**Review by CossackDex — APPROVED** [review]
